sn_networking/src/lib.rs

```diff
@@ -196,7 +200,19 @@ impl SwarmDriver {
         )?;
 
         // Listen on the provided address
+        #[cfg(not(feature = "quic"))]
         let addr = Multiaddr::from(addr.ip()).with(Protocol::Tcp(addr.port()));
+        #[cfg(not(feature = "quic"))]
+        let _listener_id = swarm_driver
+            .swarm
+            .listen_on(addr)
+            .expect("Failed to listen on the provided address");
+
+        #[cfg(feature = "quic")]
+        let addr = Multiaddr::from(addr.ip())
+            .with(Protocol::Udp(addr.port()))
+            .with(Protocol::QuicV1);
+        #[cfg(feature = "quic")]
         let _listener_id = swarm_driver
             .swarm
             .listen_on(addr)
```

Comment:
Another nit pick, we can remove the cfg for the `listen_on` call here?
```suggestion
        #[cfg(not(feature = "quic"))]
        let addr = Multiaddr::from(addr.ip()).with(Protocol::Tcp(addr.port()));

        #[cfg(feature = "quic")]
        let addr = Multiaddr::from(addr.ip())
            .with(Protocol::Udp(addr.port()))
            .with(Protocol::QuicV1);
        let _listener_id = swarm_driver
            .swarm
            .listen_on(addr)
            .expect("Failed to listen on the provided address");
```